es6-ify, linting and monster dependency update

mtharrison opened the pull request:
Quite a few changes here so probably needs another careful eye to look over it all.

nlf commented:
no one merge this until i have a chance to update the server the site runs on

nlf commented:
also @mtharrison i feel like it's reasonable to update dependencies to all their latest versions as part of this effort. thoughts?

AdriVanHoudt commented on lib/github.js:
```diff
@@ -30,31 +36,28 @@ exports.methods.push({
-            });
-
-            callback(null, issues);
+            issues = issues.filter((issue) => !issue.pull_request);
```
any reason to remove the `hasownproperty`?

mtharrison replied:
It seemed redundant.

Marsup replied:
Not sure about that, it's not the same meaning.

mtharrison replied:
It's not the same meaning but can you see any instance where this will produce incorrect behaviour?

Marsup replied:
Reading github docs again, it's an object, so :+1:

mtharrison commented:
Ok, so I've:

- Updated all deps
- Updated codebase to work with new versions of deps
- Updated tutorials for new linting config

## Normal

| Package       | Old   | New    |
|---------------|-------|--------|
| boom          | 2.x.x | 3.x.x  |
| catbox-memory | 1.x.x | 2.x.x  |
| getconfig     | 2.x.x | 3.x.x  |
| good     | 6.x.x | 7.x.x  |
| good-console     | 5.x.x | 6.x.x  |
| hapi          | 8.x.x | 13.x.x |
| hoek          | 2.x.x | 4.x.x  |
| wreck         | 6.x.x | 7.x.x  |

## Dev

| Package            | Old    | New    |
|--------------------|--------|--------|
| eslint             | 1.0.x  | 2.9.x  |
| eslint-config-hapi | 1.x.x  | 9.x.x  |
| stylus             | 0.52.x | 0.54.x |

## New

| Package | Version |
|---------|---------|
| inert   | 3.x.x   |
| vision  | 4.x.x   |

mtharrison commented on .eslintrc:
```diff
@@ -3,5 +3,9 @@
-    ]
+    ],
+    "rules": {
+        "strict": 0,
```
These rules are turned off because they don't play well with tutorial mini-snippets

AdriVanHoudt replied:
can't you add `/*eslint strict:0*/` above the tutorial pages? Or is that to much work?

mtharrison replied:
There's an .eslintrc file under /lib/tutorials, I'll add them there instead.

mtharrison commented on lib/changelog.js:
```diff
@@ -21,34 +23,31 @@ var internals = {
             staleIn: Utils.fifteenMinutes,
-            staleTimeout: 1000
+            staleTimeout: 1000,
+            generateTimeout: Utils.oneMinute
```
This seemed like a reasonable timeout, but I'm not sure. Any ideas?

mtharrison commented on server.js:
```diff
@@ -40,14 +40,15 @@ const plugins = [
-                }
-            }]
+            ops: {
+                interval: 30 * 1000
```
Not sure what interval is reasonable for this, I don't know how logs are managed/rotated on the server. Took a punt on 30s.

mtharrison commented on server.js:
```diff
@@ -41,34 +36,50 @@ server.method(require('./lib/markdown').methods);
+        register: require('good'),
+        options: {
+            ops: {
+                interval: 30 * 1000
```
Not sure what interval is reasonable for this, I don't know how logs are managed/rotated on the server. Took a punt on 30s.

AdriVanHoudt commented on lib/tutorials/auth.md:
```diff
@@ -51,7 +56,11 @@ server.register(Basic, (err) => {
+
+        if (err) {
+            throw err;
+        }
```
newline after if block?

AdriVanHoudt commented on lib/tutorials/validation.md:
````diff
@@ -151,7 +151,7 @@ For instance, a `POST` route may return one of the following:
 
 hapi supports this by allowing you to specify a different validation schema for each response status code.
 `response.status` is an object with keys that are numeric status codes, and properties that are joi schemas:
-```javascript
````
why remove the `javascript`?

mtharrison replied:
It wouldn't lint otherwise:

```
/.../tutorials/validation.md
  158:16  error  Parsing error: Unexpected token :
```

Any ideas?

AdriVanHoudt replied:
not really, it is the linter? Maybe the linter starts linting on ``` and then can't deal with a word behind it?

mtharrison replied:
I don't think it's that. All the other ```javascript blocks work fine. It doesn't seem to like the floating, non-named object.

AdriVanHoudt replied:
ah that could be it maybe ```json works?

mtharrison replied:
No, doesn't work either.

Marsup commented:
@nlf to have a suitable review, are you planning on running on node v4 or v6 ?

nlf commented:
I'll probably jump it right to 6

Marsup commented on lib/changelog.js:
```diff
@@ -152,12 +153,12 @@ exports.methods.push({
-        var v1 = matches[3];
-        var v2 = matches[4];
-        var newVersion = Semver.gt(v1, v2) ? v1 : v2;
+        const repo = matches[2];
```
Just wondering, would it be more readable with destructuring ? (`const [,, repo, v1, v2] = matches`)

AdriVanHoudt replied:
personally don't think so

Marsup commented:
Many places could benefit from shorter function definitions in object literals. Good job Matt.

mtharrison commented:
Thanks @Marsup. @nlf think we can leave that for a follow-up? Feel like there's already enough in 1 PR here.

nlf commented:
i agree. i'll get the server updated and this deployed sometime this week

Marsup commented:
Ping @nlf ;)

nlf commented:
yeah i didn't do it that week :) i'll do it today so i can move us to our swanky new server

nlf commented:
fyi i disabled auto deploys for now until i get the new server up, merging this to facilitate that
